test/test_dataloader.py

```diff
@@ -153,13 +154,14 @@ class SynchronizedSeedDataset(Dataset):
 
     def __init__(self, size, num_workers):
         assert size >= num_workers
-        self.count = multiprocessing.Value('i', 0)
+        self.count = multiprocessing.Value('i', 0, lock=True)
         self.barrier = multiprocessing.Semaphore(0)
         self.num_workers = num_workers
         self.size = size
 
     def __getitem__(self, idx):
-        self.count.value += 1
+        with self.count.get_lock():
+            self.count.value += 1
         if self.count.value == self.num_workers:
```

Comment:
Shouldn't this (the read of `self.count.value`) be in the lock too?

Reply:
good point

Reply:
I've updated the PR, although it doesn't really matter because the point is that `self.barrier` blocks until all workers reach this. If `self.barrier` gets `release`d more than once, it is fine.